TextArea: Add MaxHeight for Rows with Tags

rlingineni opened the pull request:
### Summary
Set max-height to rows when using Tags with TextArea

#### What changed?

Setting the maxHeight and minHeights to be the same in the TagsWrapper, so an overflow appears.

#### Why?
We have an issue with having too many tags, and not wrapping in the TextArea. This causes the div to grow past the given area. There may be a better design, though this adds consistency to the control in respecting the rows prop

### Links

- [Jira](https://jira.pinadmin.com/secure/RapidBoard.jspa?rapidView=1936&view=planning&selectedIssue=GESTALT-6019&quickFilter=12031&issueLimit=100)
- [TDD](link to Paper doc)
- [Figma](link to Figma file)

### Checklist
- [X] Added unit and Flow Tests
- [X] Added documentation + accessibility tests
- [] Verified accessibility: keyboard & screen reader interaction
- [X] Checked dark mode, responsiveness, and right-to-left support
- [ ] Checked stakeholder feedback (e.g. Gestalt designers)

AlbertCarreras commented:
I think you might havr to update some snapshots. 

Also, can you create an example for rows in the Docs for both text and Tags.

Can you put a Before and After example in the description?


Questions, this is set to rows={2} 
![Screenshot by Dropbox Capture](https://user-images.githubusercontent.com/10593890/234050854-48c8b4c2-6f4b-4883-8668-9d2802ec007c.png)

This is 3 rows right now 
![Screenshot by Dropbox Capture](https://user-images.githubusercontent.com/10593890/234051517-290e61eb-d6ef-4615-a7ba-b1a6b983a7a1.png)


Do we want to show 2 full lines of Tags and a little bit of the third? How should we display them? Right now

rlingineni commented:
I've updated the snapshot.

Re: 2 lines and 3 lines, may actually be a bigger change, because everywhere this is used today, the row height with tags might grow in size. 

I think it should be okay to leave it as is, and the consumer can add an extra row if they want. 

I've added a `with rows` example to the docs. Let me know!

dangerismycat commented on docs/pages/web/textarea.js:
```diff
@@ -635,6 +638,36 @@ function Example(props) {
+        <MainSection.Subsection
+          title="With rows"
+          description={`
+  You can use the rows property to set the max height of the Textarea. It will begin to overflow after the rows limit has been reached`}
```
This doesn't really capture that `rows` sets both the min and max height. Consider something like
"The `rows` prop sets the number of rows shown in `TextArea`. The input will show a scrollbar if the content exceeds the `rows` limit."

dangerismycat commented on docs/pages/web/textarea.js:
```diff
@@ -635,6 +638,36 @@ function Example(props) {
+    </Box>
+    <TextArea
+      onChange={({value})=>{setValue(value)}}
+      placeholder={"this text area has " + rows + " rows"}
```
This should use `label` so it's still visible if the user types something

dangerismycat commented on docs/pages/web/textarea.js:
```diff
@@ -635,6 +638,36 @@ function Example(props) {
+return (
+  <Box width="100%">
+    <Box marginBottom={2}>
+      <NumberField label="Number of Rows" onChange={({value})=>{setRows(value)}} value={rows}/>
```
![Screen Shot 2023-04-25 at 6 04 49 PM](https://user-images.githubusercontent.com/12059539/234439927-3c37c26e-30f2-4a94-99c8-5d3127ceaba7.png)

The UI looks a bit goofy having this NumberField the full width of the TextArea. Consider limiting this width to something that would accommodate two digits or so.

dangerismycat commented on docs/pages/web/textarea.js:
```diff
@@ -635,6 +638,36 @@ function Example(props) {
+
+return (
+  <Box width="100%">
+    <Box marginBottom={2}>
```
nit: use Flex for the parent and set `gap={2}` instead of using this Box

dangerismycat reviewed (commented):
Reiterating Alberto's request for before/after images in the PR summary, please 🙂

AlbertCarreras commented on packages/gestalt/src/TextArea.js:
```diff
@@ -221,6 +221,7 @@ const TextAreaWithForwardRef: AbstractComponent<Props, HTMLTextAreaElement> = fo
 
   const tagsWrapperStyle = {
     minHeight: rows * ROW_HEIGHT + INPUT_PADDING_WITH_TAGS,
```
I'm curious about why we need both minHeight & maxHeight being the same number, aint that just the same as height?

A follow up would be calculating the space that a row of tag + top/bottom margin takes and add ~1/3 of the next row. I think we should show rows + 1/3 of the next. But that's a design design too. We can iterate on that.

AlbertCarreras reviewed (approved):
R+, but read and address Ryan's comments
